### PR TITLE
[JENKINS-23378] Servlet 3.1 support

### PR DESCRIPTION
**Pull request by kohsuke**
Test harness now runs with Jetty 9.2

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,9 +88,14 @@ THE SOFTWARE.
...
-      <artifactId>jetty</artifactId>
-      <version>6.1.26</version>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-webapp</artifactId>
+      <version>9.2.15.v20160210</version>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-security</artifactId>
+      <version>9.2.15.v20160210</version>
```
Comment (jglick) [inline pom.xml]: Introduce a POM property.

**Comment by jglick** [comment]
:bee:

**Comment by kohsuke** [comment]
In trying this out with Jenkins 2.0 branch, I discovered that some of the tests, such as `UpdateSiteTest`, directly manipulate Jetty to create additional endpoint.

While most of the servlet 3.1 change is targeted to the 2.0 branch, I think I should integrate this new test harness to the jenkinsci/jenkins master branch as well.

**Comment by jglick** [comment]
> I think I should integrate this new test harness to the jenkinsci/jenkins master branch as well.

Unless you develop _this_ on a branch too, you have no choice, since we may well need unrelated `master`-branch changes of `jenkins-test-harness` released and integrated.

**Comment by jglick** [comment]
> some of the tests, such as `UpdateSiteTest`, directly manipulate Jetty to create additional endpoint.

This should be discouraged. https://github.com/jenkinsci/http-request-plugin/pull/13 shows how to use an unrelated server in a more test-friendly fashion.

**Pull request merged**